File: F90_src/problem.f90
!*************************************
!	POWELL FUNCTION
!*************************************
subroutine setdim(n)
	!set problem dimension n
	implicit none
	integer		:: n

	n = 4

	return
end subroutine setdim

!*************************************
subroutine setbounds(n,lb,ub)
	!set problem dimension n
	implicit none
	integer		:: n
	real*8		:: lb(n), ub(n)

	lb = - 10.d0
	ub =   10.d0

	return
end subroutine setbounds

!*************************************
subroutine startp(n,x)
	!set problem dimension n
	implicit none
	integer		:: n
	real*8		:: x(n)

	x = 1.d0

	return
end subroutine startp

!*************************************
SUBROUTINE FUNCT(N,X,F)
	implicit none
	integer		:: n
	real*8		:: x(n), f
	real*8		:: A, B, C, D

	A=X(1)+10.D0*X(2)
	B=X(3)-X(4)
	C=X(2)-2.D0*X(3)
	D=X(1)-X(4)
	F=A**2+5.D0*B**2+C**4+10.D0*D**4

	RETURN
END SUBROUTINE FUNCT

File: C_src/problem.c
#include <math.h>

void setdim(int *n){
	*n = 4;
	return;
};

void setbounds(int n, double* lb, double* ub){
	int i;
	for(i=0;i < n; i++){
		lb[i] = -10.0;
		ub[i] =  10.0;
	}
	return;
};

void startp(int n, double* xin){
	int i;
	for(i=0;i < n; i++){
		xin[i] = 1.0;
	}
	return;
}

double funct(int n, double* x){
	int i;
	double f, A, B, C, D;

	A=x[0]+10.0*x[1];
	B=x[2]-x[3];
	C=x[1]-2.0*x[2];
	D=x[0]-x[3];
	
	f=pow(A,2) + 5.0*pow(B,2) + pow(C,4) + 10.0*pow(D,4);	

	return(f);
}

